src/OrleansProviders/Streams/Common/PooledCache/ICacheDataAdapter.cs

```diff
@@ -59,6 +60,24 @@ public interface ICacheDataAdapter<in TQueueMessage, TCachedMessage>
         /// Assignable purge action.  This is called when a purge request is triggered.
         /// </summary>
         Action<IDisposable> PurgeAction { set; }
+
+        /// <summary>
+        /// Handler for when a cachedMessage got purged
+        /// </summary>
+        /// <param name="cachedMessage"></param>
+        void SignalMessagePurged(ref TCachedMessage cachedMessage);
+
+        /// <summary>
+        /// Determine action to make on a purgeRequest
+        /// </summary>
+        /// <param name="purgeRequest"></param>
+        void ConsultPurgeRequest(IDisposable purgeRequest);
```

Comment:
well, the point is not just to get an answer in this thread, but to improve either the method name to make it less ambiguous or at least make the code comment clearer (I'd prefer both, although I understand coming up with a good name is also hard)